include/functions_entries.inc.php

```diff
@@ -1113,17 +1131,19 @@ function serendipity_printEntries($entries, $extended = 0, $preview = false, $sm
     if ($use_grouped_array === false) {
         // Use grouping by date (default)
         $dategroup = array();
-        for ($x = 0, $num_entries = count($entries); $x < $num_entries; $x++) {
-            if (!empty($entries[$x]['properties']['ep_is_sticky']) && serendipity_db_bool($entries[$x]['properties']['ep_is_sticky'])) {
-                $entries[$x]['is_sticky'] = true;
-                $key = 'sticky';
-            } else {
-                $key = date('Ymd', serendipity_serverOffsetHour($entries[$x]['timestamp']));
-            }
```

Comment:
This change leads to an error in line 1336, where `$num_entries` is still referenced. The old code produced a warning when an entry was stickied?

Reply:
Yes, because `$entries[$x]` only works when it's a normal array. Sometimes there are ´'id'´s in the array. Is `$num_entries` one less in that case?

Reply:
`$num_entries` is simply not defined now. Or do you mean: Is it one less when there are ids in the array?

Reply:
Yes, for example when you have
`array(0 => 42, 1 => 144000, 2 => -1/12, 'id' => 777)`
then
´$num_entries == 3´?

Reply:
I'll check that against the old code

Reply:
Hm, I'm not able to reproduce that error. On my dev blog, `$entries` only has numeric keys. Do you remember in which situation exactly this failed?

Reply:
No, I don't remember anymore. I fixed all errors that occurred.
Now I fixed this code with ´$num_entries = count($entries)`, subtracting one if ´id´ was set.